### PR TITLE
completed final

### PR DESCRIPTION
**Pull request by WiltonCam**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -28,7 +28,19 @@ const exampleMovies = require("./movies");
       "James and the Giant Peach",
     ];
  */
-function getAllMovieTitles() {}
+function getAllMovieTitles(movies) {
+  if(movies.length === 0){
+    return [];
+  }
```
Comment (mckiernantim) [inline index.js:34]: how can we change line 32 to handle empty objects or falsy values?

File: index.js [hunk 2]
```diff
@@ -41,7 +53,21 @@ function getAllMovieTitles() {}
...
+
+  let highestScore = 0;
+  for(movie of movies){
+    let metascore = parseInt(movie.metascore)
+    if(highestScore < metascore){
+      highestScore = metascore
+    }
+  }
+
+  return highestScore;
```
Comment (mckiernantim) [inline index.js:69]: Nice work.  Remember to add `;` at the end of our statements it helps the eye know when something is done.  Also start to spread out your `{` after closing parenthesis.  Last thing I'd remove the whitespace in this function.

File: index.js [hunk 3]
```diff
@@ -105,7 +174,21 @@ function findById() {}
...
+  let arr = []
+
+  for(movie of movies){
+    let currGenres = movie.genre.toLowerCase()
+    if (currGenres.includes(genre.toLowerCase())){
+      arr.push(movie)
+    }
+  } 
+  return arr
+}
```
Comment (mckiernantim) [inline index.js:191]: great work with the .`toLowerCase()` - would this work if we used `toUpperCase()` instead?

File: index.js [hunk 4]
```diff
@@ -129,7 +212,22 @@ function filterByGenre() {}
...
+  }
+
+  let arr = []
+
+  for (movie of movies){
+    let movieYear = Number(movie.released.slice(7))
+    if(movieYear <= year){
+      arr.push(movie)
+    }
+  }
```
Comment (mckiernantim) [inline index.js:227]: This technically works but be careful with the hardcoded `.slice()`.  What if we had a data set that could be written "Twenty two November, 1976"? 
This is an edge case but its something to consider even though our tests are not asking us to consider it,

File: index.js [hunk 5]
```diff
@@ -142,7 +240,26 @@ function getAllMoviesReleasedAtOrBeforeYear() {}
...
+    let boxOfficeAmount = movie.boxOffice.slice(1)
+    let boxOfficeAmountNoCommas = boxOfficeAmount.replace(/,/g, '');
+    let num = Number(boxOfficeAmountNoCommas)
+    if (num > highestAmount){
+      highestAmount = num
+      name = movie.title
+    }
+  }
+
+  return name
```
Comment (mckiernantim) [inline index.js:261]: Nice work with the `regex`.  Could you add a `$` in there to completely sanitize this text?

**Review by mckiernantim — COMMENTED** [review]
Wilton, great work.  You've made a ton of improvement in these last weeks. Way to go.  My only advice is lets get in the habit of using `;` and watch your whitespace.  Whitespace is ok to leave in code but only in complex logic breaks which - even though this was no easy test - we don't quite have enough of for these functions.
